### ISSUE TITLE
[BUG] pyarrow.lib.ArrowInvalid Exception on a string column having `None` values returned by cat api

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#4361]
**Describe the bug**
This was tested with the new implementation on top of this PR: #4311 

For the `.str.cat` API, the underlying libcudf API returns a `unique_ptr[column]` and in cython plumbing we construct a column out of it using `Column.from_unique_ptr` and hand it off to the python side. On the python side, a StringColumn is being received and a Series is being constructed but when I try printing that series I get an exception in `to_arrow`. Below is the stack trace. 

One observation is that this exception occurs to only StringColumn which has `None` in it and being returned from the cat libcudf api. But doesn't occur when the StringColumn returned is not having any None values. Not sure if this is an issue on libcudf side or newly introduced changes to `to_arrow` function in #4336 or bridge between new changes and libcudf side are broken in this scenario, hence opening it up for discussion.

**Steps/Code to reproduce bug**

```python
>>> import cudf
>>> s = cudf.Series(['nOPq', None, 'RsT', None, 'uVw'])
>>> x = s.str.cat(others=['1', '2', '3', '4', '5'], sep=None, na_rep='')
>>> x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/series.py", line 565, in __repr__
    output = preprocess.to_pandas().__repr__()
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/series.py", line 1386, in to_pandas
    s = self._column.to_pandas(index=index)
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/column/string.py", line 1408, in to_pandas
    pd_series = self.to_arrow().to_pandas()
  File "/conda/envs/cudf/lib/python3.7/site-packages/cudf/core/column/string.py", line 1404, in to_arrow
    len(self), obuf, sbuf, nbuf, self.null_count
  File "pyarrow/array.pxi", line 1363, in pyarrow.lib.StringArray.from_buffers
  File "pyarrow/array.pxi", line 724, in pyarrow.lib.Array.from_buffers
  File "pyarrow/array.pxi", line 976, in pyarrow.lib.Array.validate
  File "pyarrow/error.pxi", line 78, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Offset invariant failure at: 2 inconsistent value_offsets for null slot6!=5
>>> x._column.children[0]
<cudf.core.column.numerical.NumericalColumn object at 0x7fcf6fbf3550>
>>> x._column.children[1]
<cudf.core.column.numerical.NumericalColumn object at 0x7fcf6fbf3410>
>>> x._column.children[0].data_array_view.copy_to_host()
array([ 0,  5,  6, 10, 11, 15], dtype=int32)
>>> x._column.children[1].data_array_view.copy_to_host()
array([110,  79,  80, 113,  49,  82,  82, 115,  84,  51,  86, 117,  86,
       119,  53], dtype=int8)



Working scenario
>>> s = cudf.Series(['nOPq', 'x' , 'RsT', 'x', 'uVw'])
>>> s.str.cat(others=['1','2','3','4','5'], na_rep=None)
0    nOPq1
1       x2
2     RsT3
3       x4
4     uVw5
dtype: object
```

**Expected behavior**
The expectation is that string concatenation happens after replacing None with whatever `na_rep` value is being provided.

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source

**Environment details**
output of the `cudf/print_env.sh` script here: [env.txt](https://github.com/rapidsai/cudf/files/4296029/env.txt)

**Additional context**
None

**Comment by jrhemstad** [comment]
So is this a problem with calling `to_arrow` on any Strings column? Or only for Strings columns that result from `cat`?

**Comment by kkraus14** [comment]
> So is this a problem with calling `to_arrow` on any Strings column? Or only for Strings columns that result from `cat`?

So far we've only seen it for String columns resulting from `cat`, but specifically it's a problem with String Columns that have non 0 byte null elements, which Arrow says are invalid.

**Comment by kkraus14** [comment]
Fixed by #4358

**Comment by davidwendt** [comment]
@galipremsagar  Just checking with you that this is fixed by #4358
